coala_quickstart/green_mode/green_mode_core.py

```diff
@@ -0,0 +1,80 @@
+    structure of the project, runs the QuickstartBear which guesses some values
+    of settings the can take an infinite set of values by parsing the
+    file_dict and appends to `.project_data.yaml`. Runs some further linting
+    options based on file names etc. Calls the methods which test out whether
+    a setting value is green for a bear i.e. does not point out any error in
+    the code base and further generates sections and writes the green config
+    file for the project.
+    :param project_dir:
+        The project directory.
+    :param ignore_globs:
+        The globs of the files to ignore from the linting process.
+    :param bears:
+        The bears from Constants/GREEN_MODE_COMPATIBLE_BEAR_LIST along
+        with Constants/IMPORTANT_BEAR_LIST.
+    :param bear_settings_obj:
+        The object of SettingsClass/BearSettings which stores the metadata
+        about whether a setting takes a boolean value or any other value.
+    :param op_args_limit:
+        The maximum number of optional bear arguments allowed for guessing.
+    :param value_to_op_args_limit:
+        The maximum number of values to run the bear again and again for
+        a optioanl setting.
+    """
+    from coala_quickstart.green_mode.filename_operations import (
+        check_filename_prefix_postfix)
+    ignore_globs.append(project_dir+os.sep+'.git'+os.sep+'**')
+    project_data = project_dir + os.sep + '.project_data.yaml'
+
+    # Currently as a temporary measure, recreating the file at each run from
+    # scratch, as there is no mechanism created uptil now to reuse this data.
+    if os.path.isfile(project_data):
+        os.remove(project_data)
+
+    if not os.path.isfile(project_data):
+        new_data = initialize_project_data(project_dir+os.sep, ignore_globs)
+        data_to_dump = {'dir_structure': new_data}
+        dump_to_file(project_data, data_to_dump)
+    else:
+        pass
+        # Add steps to reuse this data and modify this file later on.
+
+    # Operations before the running of QuickstartBear are done over here.
+    # Eg. do operations on filenames over here.
+    project_data_contents = get_contents(project_data)
+    project_data_contents = check_filename_prefix_postfix(
+        project_data_contents)
+
+    # Run QuickstartBear
+    (project_data_contents, ignore_ranges, file_dict,
+     file_names) = run_quickstartbear(
+        project_data_contents, project_dir)
+
+    final_non_op_results, final_unified_results = bear_test_fun(
+        bears, bear_settings_obj, file_dict,
+        ignore_ranges, project_data_contents, file_names,
+        op_args_limit, value_to_op_args_limit, printer)
+
+    # Call to create `.coafile` goes over here.
+
+    # Final Dump.
+    dump_to_file(project_data, project_data_contents)
+
+    # Delete .project_data.yaml for now as there is currently no mechanism
+    # added to reuse this data.
+    os.remove(project_dir+os.sep+'.project_data.yaml')
```

Comment:
Instead of doing `project_dir+os.sep+'.project_data.yaml'` everytime you can create a single variable and reuse it.

Reply:
Also, I was wondering if we can add `'project_data.yaml' to a constant. Makes it easier if we wish to change the name of this file later on.